File: examples/lib/src/androidTest/java/com/github/neboskreb/log4j2/examples/lib/AwesomeWorkerAndroidTest.java
package com.github.neboskreb.log4j2.examples.lib;

import net.loune.log4j2android.AndroidLog4jExtension;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.junit.jupiter.api.Assertions.assertEquals;

@ExtendWith(AndroidLog4jExtension.class)
public class AwesomeWorkerAndroidTest {

    @Test
    public void doGreatJob() {
        // GIVEN
        AwesomeWorker AwesomeWorker = new AwesomeWorker();

        // WHEN
        String result = AwesomeWorker.doGreatJob("foo", "bar");

        // THEN
        assertEquals("foobar", result);
    }

    @Test
    public void doNotSoGreatJob() {
        // GIVEN
        AwesomeWorker AwesomeWorker = new AwesomeWorker();

        // WHEN
        AwesomeWorker.doNotSoGreatJob();

        // THEN no exceptions thrown
    }
}

File: examples/lib/src/androidTest/kotlin/com/github/neboskreb/log4j2/examples/lib/AwesomeWorkerAndroidTest.kt
package com.github.neboskreb.log4j2.examples.lib

import com.github.neboskreb.log4j2.android.AndroidLog4jExtension
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

@ExtendWith(AndroidLog4jExtension::class)
class AwesomeWorkerAndroidTest {
    @Test
    fun doGreatJob() {
        // GIVEN
        val AwesomeWorker = AwesomeWorker()

        // WHEN
        val result = AwesomeWorker.doGreatJob("foo", "bar")

        // THEN
        Assertions.assertEquals("foobar", result)
    }

    @Test
    fun doNotSoGreatJob() {
        // GIVEN
        val AwesomeWorker = AwesomeWorker()

        // WHEN
        AwesomeWorker.doNotSoGreatJob()

        // THEN no exceptions thrown
    }

    @Test
    fun staticLogging() {
        // WHEN
        logSomething()

        // THEN see logs
    }
}
